test/World_newTests/WhichRobotHasBallTest.cpp

```diff
@@ -41,40 +41,27 @@ TEST(World_newTest, WhichRobotHasBallTest) {
     robotsYellow.Add()->CopyFrom(roboty2);
     robotsBlue.Add()->CopyFrom(robotb1);
 
-    /** Test 1 : no ball. No robot closest */
-    proto::World msgNoBall;
-    msgNoBall.mutable_yellow()->CopyFrom(robotsYellow);
-    msgNoBall.mutable_blue()->CopyFrom(robotsBlue);
-    worldInstance->updateWorld(msgNoBall);
-
-    auto world = worldInstance->getWorld();
-    assert(world->getUs().size() == 2);
-    assert(world->getThem().size() == 1);
-    assert(!world->getBall().has_value());
-    std::optional<rtt::world_new::view::RobotView> robot = world->whichRobotHasBall(rtt::world_new::Team::us);
-    EXPECT_TRUE(!robot.has_value());
-
-    /** Test 2 : Two yellow robots (1 & 2), two blue robots (3 & 4), one ball **/
```

Comment:
Why was this removed?

Reply:
We always have a ball now whenever we create a ball from a proto message after the fix in Ball.cpp. (the ball staying with a robot when there is no vision)